mjtest-files/compile/bench/AbstractedMath.mj

```diff
@@ -0,0 +1,87 @@
+/* A really inefficient way to calculate powers of two.
+   We use a math library to hide the actual calculations
+   under a tower of abstractions. Hopefully, some of this
+   can be inlined or unrolled...
+*/
+class FindPowersOfTwo {
+    public static void main(String[] args) {
+        FindPowersOfTwo obj = new FindPowersOfTwo();
+        obj.run();
+    }
+
+    public void run() {
+        int currentLog = -1;
+        int i = 1;
+        while (i < 10000000) {
+            Math math = new Math();
+            if (math.log(2, i) > currentLog) {
+                currentLog = math.log(2, i);
+                System.out.println(i);
+            }
+            i = i + 1;
+        }
+    }
+}
+
+/* A math library that reimplements basic operations. At least logarithmic time, always! */
+class Math {
+    /* trial and error logarithm, rounded up */
+    public int log(int base, int val) {
+        int sign = 1;
+        if (val < 0) {
+            sign = -1;
+            val = -val;
+        }
+        int sum = 0;
+        int guess = 5;
+        while (val / pow(base, guess) > 1) {
+            sum = sum + guess;
```

Comment:
needs some refactoring to a method that performs an addition, to make code more readable

Reply:
Don't know. A compiler without an inliner doesn't have the slightest chance here, anyway